src/mumble/AudioOutput.h

```diff
@@ -121,14 +120,16 @@ class AudioOutput : public QThread {
 	~AudioOutput() Q_DECL_OVERRIDE;
 
 	void addFrameToBuffer(ClientUser *sender, const Mumble::Protocol::AudioData &audioData);
-	void removeBuffer(const ClientUser *);
-	AudioOutputSample *playSample(const QString &filename, float volume, bool loop = false);
+	std::unique_ptr< AudioOutputToken > playSample(const QString &filename, float volume, bool loop = false);
```

Comment:
This return type doesn't make sense. This should return `AudioOutputToken` by value. As it stands, this code essentially returns a pointer object that itself has been allocated on the heap and therefore you now get a pointer to a pointer to an object....